lib/banks/checking.rb

```diff
@@ -0,0 +1,17 @@
+require "pry"
+module BankAccount
+  class CheckingAccount < Account
+      MINIMUM_BALANCE = 0
+      TRANSACTION_FEE = 1
+
+    def withdraw(amount)
+      # This method inherits from the Account class #withdraw(amount) method
+      # amount will be checked against current_balance to make sure you have
+      # enough money in your account to make this withdrawl (leaving at least
+      # $10 left in the account)
+      amount = amount + MINIMUM_BALANCE + TRANSACTION_FEE
+      # need to add MINIMUM_BALANCE back on the current_balance
+      super + MINIMUM_BALANCE
```

Comment:
I was pretty confused when I saw this method the first time, but I dig what you're doing. You're manipulating the amount of the withdrawal before invoking the inherited behavior. It's kinda confusing to me because when a method invokes `super` without parens/parameters, `super` gets the same parameters as was given the inheriting method.

I read line 14 as `super` would receive the `amount` parameter _as it was passed into the `#withdraw` method_, but that's not exactly right. You mutate `amount` on line 12, and that mutated value is passed into super. Just for clarity, I think I'd write this like...

```ruby
adjusted_amount = amount + MINIMUM_BALANCE + TRANSACTION_FEE
super(adjusted_amount) + MINIMUM_BALANCE
```

That'd be more clear as to what you're giving `super` and why. All that aside, your commenting is great here and helped me understand your process.